### PR TITLE
feat: add `act_fn` param to `OutValueFunctionBlock`

### LINKED ISSUE #1287: Update UNet1D Tests to not use Mish
**Is your feature request related to a problem? Please describe.**
Currently the `UNet1DTests` heavily rely on mish because they were based on a re-implementation of a model. This is not needed generally, so it would be good to add another 1d model just for tests so we can delete a lot of the tests that look like this.

```
    @unittest.skipIf(torch_device == "mps", "mish op not supported in MPS")
    def test_determinism(self):
        super().test_determinism()
```

### PR DESCRIPTION
**Pull request by SauravMaheshkar**
# What does this PR do?

(Hopefully) fixes #1287 

## Before submitting
- [ ] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/diffusers/blob/main/CONTRIBUTING.md)?
- [x] Did you read our [philosophy doc](https://github.com/huggingface/diffusers/blob/main/PHILOSOPHY.md) (important for complex PRs)?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/diffusers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

Request for Review: @natolambert

### CODE REVIEW

File: src/diffusers/models/unet_1d_blocks.py [hunk 1]
```diff
@@ -235,12 +235,12 @@ def forward(self, hidden_states, temb=None):
 
 
 class OutValueFunctionBlock(nn.Module):
-    def __init__(self, fc_dim, embed_dim):
+    def __init__(self, fc_dim, embed_dim, act_fn):
```
Comment (patrickvonplaten) [inline src/diffusers/models/unet_1d_blocks.py:238]:
```suggestion
    def __init__(self, fc_dim, embed_dim, act_fn="mish"):
```

**Comment by patrickvonplaten** [comment]
I don't fully understand what this fixes. Could you maybe elaborate here @SauravMaheshkar ? :-)

**Comment by SauravMaheshkar** [comment]
> I don't fully understand what this fixes. Could you maybe elaborate here @SauravMaheshkar ? :-)

Hey @patrickvonplaten thanks for getting back. This doesn't fix an issue but I rather wanted to raise the discussion about how to fix #1287. A very naive fix could be to simply change the activation to `swish` in `prepare_init_args_and_inputs_for_common` ( in `tests/models/test_models_unet_1d.py`) solve the problem (and updating the `OutValueFunctionBlock` to not have `mish`)?

But if not what would you suggest is a way to fix it ? I'm afraid I can't understand https://github.com/huggingface/diffusers/issues/1287#issuecomment-1336339215

**Review by patrickvonplaten — APPROVED** [review]

**Comment by patrickvonplaten** [comment]
TBH, I also don't fully understand the discussion of #1287 - I'm ok with merging this PR as is though if it helps for your use case :-)

**Comment by SauravMaheshkar** [comment]
Well this PR changes the activation function to be `swish` instead of `mish` so we can drop the decorators (`@unittest.skipIf(torch_device == "mps", "mish op not supported in MPS")`)

**Comment by patrickvonplaten** [comment]
Perfect!

**Pull request merged**